packages/apps/reputation-oracle/server/src/common/utils/index.ts

```diff
@@ -0,0 +1,144 @@
+    }
+}
+
+
+type InputFile = {
+    type: 'path',
+    path: string,
+} | {
+    type: 'object',
+    content: any,
+};
+
+// TODO: Can be used for future implementaton in SDK Storage module
+/**
+   * **Upload file to cloud storage**
+   *
+   * @param {File[]} files - Files to upload
+   * @param {string} bucket - Bucket name
+   * @param {StorageParams} params - Cloud storage params
+   * @param {StorageCredentials} credentials - Optional. Cloud storage access data. If credentials is not provided - use an anonymous access to the bucket
+   * @returns {Promise<UploadFile>} - Uploaded file with key/hash
+   */
+export async function uploadFiles(
+    files: InputFile[],
+    bucket: string,
+    clientParams: StorageParams,
+    credentials?: StorageCredentials,
+): Promise<UploadFile[]> {
+    const client: Minio.Client = new Minio.Client({
+        ...clientParams,
+        accessKey: credentials?.accessKey ?? '',
+        secretKey: credentials?.secretKey ?? '',
+    });
+
+    const isBucketExists = await client.bucketExists(bucket);
+    if (!isBucketExists) {
+        throw new Error('Bucket not found');
+    }
+
+    return Promise.all(
+        files.map(async (fileInput) => {
+            let hash: crypto.Hash;
+            let contentType;
+            let contentLength;
+            let stream;
+
+            if (fileInput.type === 'path') {
+                const filePath = fileInput.path;
+                stream = fs.createReadStream(filePath);
+                hash = crypto.createHash('sha1');
+                stream.on('data', (data) => hash.update(data));
+
+                // Setting a default content type (can be refined)
+                contentType = filePath.endsWith('.zip') ? 'application/zip' : 'application/octet-stream';
+                contentLength = fs.statSync(filePath).size;
+            } else { // for in-memory objects
+                const content = JSON.stringify(fileInput.content);
+                stream = Buffer.from(content);
+                hash = crypto.createHash('sha1').update(content);
+                contentType = 'application/json';
+                contentLength = Buffer.byteLength(content);
+            }
+
+            const fileHash = hash.digest('hex');
+            const key = `s3${fileHash}.${contentType === 'application/zip' ? 'zip' : 'json'}`;
+
+            try {
+                await client.putObject(bucket, key, stream, {
+                    'Content-Type': contentType,
+                    'Content-Length': contentLength,
+                });
+
+                return {
+                    key,
+                    url: `${clientParams.useSSL ? 'https' : 'http'}://${
+                        clientParams.endPoint
+                    }${clientParams.port ? `:${clientParams.port}` : ''}/${bucket}/${key}`,
+                    hash: fileHash,
+                };
+            } catch (e) {
+                throw new Error('Error uploading file');
```

Comment:
Iiuc, there is no way to know which file caused the problem, right? Could we include additional information in the error message that allows to identify which file failed to upload?

```suggestion
                let message = '';
                if (fileInput.type === 'path') {
                    message = `Error uploading file from path: ${fileInput.path}`);
                else {
                    message = `Error upload in-memory file. SHA1 hash: ${fileHash}`
                }
                throw new Error(message);
```